spyder_unittest/backend/testrunner.py

```diff
@@ -0,0 +1,206 @@
+                                               'unittest.results')
+        else:
+            self.resultfilename = resultfilename
+
+    def start(self, config, pythonpath):
+        """
+        Start process which will run the unit test suite.
+
+        The process is run in the working directory specified in 'config',
+        with the directories in `pythonpath` added to the Python path for the
+        test process. The test results are written to the file
+        `self.resultfilename`. The standard output and error are also recorded.
+        Once the process is finished, `self.finished()` will be called.
+
+        Parameters
+        ----------
+        config : TestConfig
+            Unit test configuration.
+        pythonpath : list of str
+            List of directories to be added to the Python path
+
+        Raises
+        ------
+        RuntimeError
+            If process failed to start.
+        """
+
+        framework = config.framework
+        wdir = config.wdir
+
+        self.process = QProcess(self)
+        self.process.setProcessChannelMode(QProcess.MergedChannels)
+        self.process.setWorkingDirectory(wdir)
+        self.process.finished.connect(self.finished)
+
+        if pythonpath is not None:
+            env = [
+                to_text_string(_pth)
+                for _pth in self.process.systemEnvironment()
+            ]
+            add_pathlist_to_PYTHONPATH(env, pythonpath)
+            processEnvironment = QProcessEnvironment()
+            for envItem in env:
+                envName, separator, envValue = envItem.partition('=')
+                processEnvironment.insert(envName, envValue)
+            self.process.setProcessEnvironment(processEnvironment)
+
+        if framework == 'nose':
+            executable = "nosetests"
+            p_args = ['--with-xunit', "--xunit-file=%s" % self.resultfilename]
+        elif framework == 'py.test':
+            executable = "py.test"
+            p_args = ['--junit-xml', self.resultfilename]
+        else:
+            raise ValueError('Unknown framework')
+
+        if os.name == 'nt':
+            executable += '.exe'
+
+        try:
+            os.remove(self.resultfilename)
+        except OSError:
+            pass
+
+        self.process.start(executable, p_args)
+        running = self.process.waitForStarted()
+        if not running:
+            raise RuntimeError
+
+    def finished(self):
+        """
+        Called when the unit test process has finished.
+
+        This function reads the results and emits `sig_finished`.
+        """
+        qbytearray = self.process.readAllStandardOutput()
+        locale_codec = QTextCodec.codecForLocale()
+        output = to_text_string(locale_codec.toUnicode(qbytearray.data()))
+        testresults = self.load_data()
+        self.sig_finished.emit(testresults, output)
+
+    def kill_if_running(self):
+        """Kill testing process if it is running."""
```

Comment:
Kill. I changed the user interface to also use the word "kill", as this is clearer on what happens.